pkg/config/config.go

```diff
@@ -80,6 +80,34 @@ type Server struct {
 	CacheDictionarySize int               `deprecated:"true"`
 	CacheSegmentSize    int               `deprecated:"true"`
 	CacheTreeSize       int               `deprecated:"true"`
+
+	// TODO: remove skip: true when we enable these back
+	GoogleEnabled      bool   `skip:"true" def:"false" desc:"enables Google Oauth"`
+	GoogleClientID     string `skip:"true" def:"" desc:"client ID generated for Google API"`
+	GoogleClientSecret string `skip:"true" def:"" desc:"client secret generated for Google API"`
+	GoogleRedirectURL  string `skip:"true" def:"" desc:"url that google will redirect to after logging in. Has to be in form <pathToPyroscopeServer/google/callback>"`
+	GoogleAuthURL      string `skip:"true" def:"https://accounts.google.com/o/oauth2/auth" desc:"auth url for Google API (usually present in credentials.json file)"`
+	GoogleTokenURL     string `skip:"true" def:"https://accounts.google.com/o/oauth2/token" desc:"token url for Google API (usually present in credentials.json file)"`
+
+	GitlabEnabled bool `skip:"true" def:"false" desc:"enables Gitlab Oauth"`
+	// TODO: why is this one ApplicationID and not ClientID ?
+	GitlabApplicationID string `skip:"true" def:"" desc:"application ID generated for GitLab API"`
+	GitlabClientSecret  string `skip:"true" def:"" desc:"client secret generated for GitLab API"`
+	GitlabRedirectURL   string `skip:"true" def:"" desc:"url that gitlab will redirect to after logging in. Has to be in form <pathToPyroscopeServer/gitlab/callback>"`
+	GitlabAuthURL       string `skip:"true" def:"https://gitlab.com/oauth/authorize" desc:"auth url for GitLab API (keep default for cloud, usually https://gitlab.mycompany.com/oauth/authorize for on-premise)"`
+	GitlabTokenURL      string `skip:"true" def:"https://gitlab.com/oauth/token" desc:"token url for GitLab API (keep default for cloud, usually https://gitlab.mycompany.com/oauth/token for on-premise)"`
+	GitlabAPIURL        string `skip:"true" def:"https://gitlab.com/api/v4/user" desc:"URL to gitlab API (keep default for cloud, usually https://gitlab.mycompany.com/api/v4/user for on-premise)"`
+
+	GithubEnabled      bool   `skip:"true" def:"false" desc:"enables Github Oauth"`
+	GithubClientID     string `skip:"true" def:"" desc:"client ID generated for Github API"`
+	GithubClientSecret string `skip:"true" def:"" desc:"client secret generated for Github API"`
+	GithubRedirectURL  string `skip:"true" def:"" desc:"url that Github will redirect to after logging in. Has to be in form <pathToPyroscopeServer/github/callback>"`
+	GithubAuthURL      string `skip:"true" def:"https://github.com/login/oauth/authorize" desc:"auth url for Github API"`
+	GithubTokenURL     string `skip:"true" def:"https://github.com/login/oauth/access_token" desc:"token url for Github API"`
+
+	// TODO: can we generate these automatically if it's empty?
```

Comment:
We can, we would just have to write it in some kind of persistent storage so that users don't get logged out if pyroscope is restarted. It's either that or making users explicitly generate it, which I think should be a good security practice (but it does make less of the things out-of-the-box)